test/Microsoft.OpenApi.Tests/Services/OpenApiSecurityRequirementComparerTests.cs

```diff
@@ -0,0 +1,293 @@
+                        "scheme2", new OpenApiSecurityScheme
+                        {
+                            Description = "Test",
+                            Name = "Test"
+                        }
+                    },
+                    {
+                        "scheme4", new OpenApiSecurityScheme
+                        {
+                            Description = "Test",
+                            Name = "Test"
+                        }
+                    }
+                }
+            }
+        };
+
+        public OpenApiSecurityRequirementComparerTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        public static IEnumerable<object[]> GetTestCasesForOpenApiSecurityRequirementComparerShouldSucceed()
+        {
+            yield return new object[]
+            {
+                "New Removed And updated schemes",
+                new OpenApiSecurityRequirement
+                {
+                    [
+                        new OpenApiSecurityScheme
+                        {
+                            Reference = new OpenApiReference {Type = ReferenceType.SecurityScheme, Id = "scheme1"}
+                        }
+                    ] = new List<string>
+                    {
+                        "scope1",
+                        "scope2",
+                        "scope3"
+                    },
+                    [
+                        new OpenApiSecurityScheme
+                        {
+                            Reference = new OpenApiReference {Type = ReferenceType.SecurityScheme, Id = "scheme2"}
+                        }
+                    ] = new List<string>
+                    {
+                        "scope4",
+                        "scope5"
+                    },
+                    [
+                        new OpenApiSecurityScheme
+                        {
+                            Reference = new OpenApiReference {Type = ReferenceType.SecurityScheme, Id = "scheme3"}
+                        }
+                    ] = new List<string>()
+                },
+                new OpenApiSecurityRequirement
+                {
+                    [
+                        new OpenApiSecurityScheme
+                        {
+                            Reference = new OpenApiReference {Type = ReferenceType.SecurityScheme, Id = "scheme1"}
+                        }
+                    ] = new List<string>
+                    {
+                        "scope1",
+                        "scope2",
+                        "scope3"
+                    },
+                    [
+                        new OpenApiSecurityScheme
+                        {
+                            Reference = new OpenApiReference {Type = ReferenceType.SecurityScheme, Id = "scheme2"}
+                        }
+                    ] = new List<string>
+                    {
+                        "scope4",
+                        "scope5"
```

Comment:
Try add test cases with two different scope lists

Reply:
I am currently not comparing scope lists, have replied on other thread on question i have on diff of it

---
In reply to: [223517115](https://github.com/Microsoft/OpenAPI.NET/pull/328#discussion_r223517115) [](ancestors = 223517115)

Reply:
Will cover in another PR

---
In reply to: [223853512](https://github.com/Microsoft/OpenAPI.NET/pull/328#discussion_r223853512) [](ancestors = 223853512,223517115)